### PR TITLE
[RFC] Change return type to bool on diff.c functions

### PR DESCRIPTION
**Pull request by war1025**

### CODE REVIEW

File: src/nvim/diff.c [hunk 1]
```diff
@@ -1535,25 +1535,24 @@ int diff_check(win_T *wp, linenr_T lnum)
   return maxcount - dp->df_count[idx];
 }
 
-/// Compare two entries in diff "*dp" and return TRUE if they are equal.
+/// Compare two entries in diff "*dp" and return true if they are equal.
```
Comment (bfredl) [inline src/nvim/diff.c]: Could drop the '*' before `dp` (and in param below)

File: src/nvim/diff.c [hunk 2]
```diff
@@ -1834,28 +1833,30 @@ int diffopt_changed(void)
   return OK;
 }
 
-/// Return TRUE if 'diffopt' contains "horizontal".
+/// Return true if 'diffopt' contains "horizontal".
 ///
-/// @return TRUE if 'diffopt' contains "horizontal"
-int diffopt_horizontal(void)
+/// @return true if 'diffopt' contains "horizontal"
```
Comment (Pyrohh) [inline src/nvim/diff.c]: This is the exact same sentence as the opening sentence, so I think it should be removed.

File: src/nvim/diff.c [hunk 3]
```diff
@@ -1960,21 +1961,21 @@ int diff_find_change(win_T *wp, linenr_T lnum, int *startp, int *endp)
   return added;
 }
 
-/// Return TRUE if line "lnum" is not close to a diff block, this line should
+/// Return true if line "lnum" is not close to a diff block, this line should
 /// be in a fold.
 ///
-/// @param wp
-/// @param lnum
+/// @param wp   The window whose current buffer to check
```
Comment (Pyrohh) [inline src/nvim/diff.c]: maybe `The window containing the buffer to check`?

File: src/nvim/diff.c (hunk 2 above)
Comment (war1025) [inline src/nvim/diff.c]: That was my thought as well, but it seemed that people wanted "full docs" so I went all out...

I would be happy to remove them as they seem redundant to me as well.

Comment (Pyrohh) [inline src/nvim/diff.c]: maybe @bfredl can chime in.  I think we should avoid such redundancy, but don't feel strongly on it.

Comment (bfredl) [inline src/nvim/diff.c]: I looked a bit around the codebase, there is quite many pure functions (whose only purpose is to return something). Some have the same repetition, some have slightly reformulated short description like "check if XX" and then "returns true if XX" which sounds better but still basically the same redundancy. I don't know how flexible doxygen is, but I would guess the following won't work? (taking an example with a param)
```
 /// @return true if buffer "buf" is in diff-mode.
 ///
 /// @param buf The buffer to check.
 bool diff_mode_buf(buf_T *buf)
```
If not I suppose we could just drop the `@return` in those cases.

**Comment by bfredl** [comment]
LGTM after existing comments
also a rebase is needed

**Pull request closed without merging**